src/marks/axis.js

```diff
@@ -652,7 +652,7 @@ function inferTextChannel(scale, data, ticks, tickFormat, anchor) {
 // possible, or the default ISO format (2014-01-26). TODO We need a better way
 // to infer whether the ordinal scale is UTC or local time.
 export function inferTickFormat(scale, data, ticks, tickFormat, anchor) {
-  return typeof tickFormat === "function"
+  return typeof tickFormat === "function" && !(scale.type === "log" && scale.tickFormat)
```

Comment:
```suggestion
  return typeof tickFormat === "function" && scale.type !== "log"
```

Nit: I'm not sure about the case for the second part of the test. When scale.type is "log", the scale is an instance of d3.scaleLog() and it has a tickFormat method, which the user can't nullify.

Reply:
I didn’t do this because I didn’t want to assume that log scales necessarily implement `scale.tickFormat`. That is currently true, but if this assumption doesn’t hold in the future, this could would break.